elements/campaign-display/components/campaign-display-root.js

```diff
@@ -6,11 +6,11 @@ export default function CampaignDisplayRoot (props) {
   let component;
   switch (props.display) {
   case 'name':
-    component = <CampaignDisplayName name={props.data.name}/>;
+    component = <CampaignDisplayName name={props.campaign.name}/>;
     break;
 
   case 'image':
-    component = <CampaignDisplayImage name={props.data.image} />;
+    component = <CampaignDisplayImage {...props.campaign} />;
```

Comment:
Actually, no. I don't think there's any value in manually unpacking the attributes. This is a common pattern for react and I don't see that it's an issue. This is what happens with a wrapper component which just delegates to more specific components.